src/renderers/dom/client/ReactMount.js

```diff
@@ -439,6 +440,10 @@ var ReactMount = {
       null,
       nextElement
     );
+    var nextContext = parentComponent ?
+      parentComponent._reactInternalInstance._processChildContext(
```

Comment:
Can you change this to `ReactInstanceMap.get(parentComponent)`? Only that file should have `_reactInternalInstance` hardcoded.